Core/include/Acts/Detector/detail/IndexedSurfacesGenerator.hpp

```diff
@@ -59,7 +59,7 @@ struct IndexedSurfacesGenerator {
   /// @param rGenerator the reference generataor
   ///
   /// @return a SurfaceCandidateUpdator delegate
-  template <typename axis_generator, typename reference_generator>
+  template <typename axis_generator, typename reference_generator, template<typename> class IndexedGridSurfacesUpdator>
   SurfaceCandidatesUpdator operator()(
```

Comment:
just for me to understand - where do you need this functionality?

Reply:
Here, the `IndexedSurfacesImpl` is only used as navigation delegeate for indexed surfaces, as I understand. I would like to make it as we can pass other delegate types for indexed surfaces (e.g. I am working on a delegate where specific indexed surfaces will be assigned to candidates by checking with intersection and not assign all the surfaces of the volume).